src/restruct_poc/model/model.hpp

```diff
@@ -15,40 +15,40 @@ struct Model {
     std::vector<Constraint> constraints;
 
     const Node AddNode(
```

Comment:
You're right, a regular node would suffice. I was thinking that it would indicate that the node values shouldn't be changed, but it doesn't really matter. Eventually I'd like for these to return a reference or pointer to the nodes in the vector so they can be manipulated, but I need some help figuring out how to do that.